lib/common/socket.c

```diff
@@ -828,6 +833,18 @@ static void on_handshake_complete(h2o_socket_t *sock, const char *err)
         }
     }
 
+    if (sock->ssl->handshake.client.session_cache) {
```

Comment:
I'd appreciate if the `if` statement was written to evaluate a boolean value, e.g. by comparing a pointer against `!= NULL`.